pkg/services/services.go

```diff
@@ -193,12 +189,57 @@ func klogVerbosityArg(debug bool) string {
 }
 
 // getKubeletHostnameOverride returns the hostname override arg that should be used
-func getKubeletHostnameOverride(platformType string) string {
+func getKubeletHostnameOverride(platformType string) (string, servicescm.PowershellCmdArg) {
 	platformType = strings.ToUpper(platformType)
+	// define argument with fixed variable name
+	hostnameOverrideArg := "--hostname-override=" + hostnameOverrideVarName
 	switch platformType {
 	case string(config.AWSPlatformType):
-		return "--hostname-override=" + ec2HostnameVar
+		return hostnameOverrideArg, getAWSMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
+	case string(config.GCPPlatformType):
+		return hostnameOverrideArg, getGCPMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
 	default:
-		return ""
+		return "", servicescm.PowershellCmdArg{}
 	}
 }
+
+// getAWSMetadataHostnamePowershellCmdArg returns the PowershellCmdArg to resolve the hostname using the instance
+// metadata service in AWS. Uses AWS Tools for PowerShell (https://docs.aws.amazon.com/powershell/latest/userguide/pstools-welcome.html)
+func getAWSMetadataHostnamePowershellCmdArg(variableName string) servicescm.PowershellCmdArg {
+	return servicescm.PowershellCmdArg{
+		Name: variableName,
+		Path: "Get-EC2InstanceMetadata -Category LocalHostname",
+	}
+}
+
+// getGCPMetadataHostnamePowershellCmdArg returns the PowershellCmdArg to resolve the hostname using the instance
+// metadata service in GCP. Uses GoogleCloud Tools for PowerShell (https://cloud.google.com/tools/powershell/docs/quickstart)
+// getGCPMetadataHostname returns the GCP instance hostname from the metadata service.  GCP is an especial case,
+// and the resulting hostname is truncated if the length is longer than 63 characters.
+func getGCPMetadataHostnamePowershellCmdArg(variableName string) servicescm.PowershellCmdArg {
+	return servicescm.PowershellCmdArg{
+		Name: variableName,
+		Path: `
+# MAX_LENGTH is the maximum number of character allowed for the instance's hostname in GCP
```

Comment:
Is this/should this be doc'd somewhere so no one accidentally creates an instance more than 63 chars?

Reply:
The proposed implementation will take care of this so that the naming is consistent, even if the length of the instance's name is greater than 63 chars.